conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -940,38 +948,101 @@ public Boolean visitUnknown(String unknownType) {
+                }
+                return value.getItemType().accept(new DefaultTypeVisitor<>() {
+                    @Override
+                    public CollectionType visitDefault() {
+                        return new CollectionType(
+                                ConjureCollectionType.LIST,
+                                ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                    }
+
+                    @Override
+                    public CollectionType visitPrimitive(PrimitiveType primitiveType) {
+                        return primitiveType.accept(new DefaultPrimitiveTypeVisitor<>() {
+
+                            @Override
+                            public CollectionType visitDefault() {
+                                return new CollectionType(
+                                        ConjureCollectionType.LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+
+                            @Override
+                            public CollectionType visitDouble() {
+                                return new CollectionType(
+                                        ConjureCollectionType.DOUBLE_LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+
+                            @Override
+                            public CollectionType visitInteger() {
+                                return new CollectionType(
+                                        ConjureCollectionType.INTEGER_LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+
+                            @Override
+                            public CollectionType visitBoolean() {
+                                return new CollectionType(
+                                        ConjureCollectionType.BOOLEAN_LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+
+                            @Override
+                            public CollectionType visitSafelong() {
+                                return new CollectionType(
+                                        ConjureCollectionType.SAFE_LONG_LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+                        });
+                    }
+                });
             }
 
             @Override
             public CollectionType visitSet(SetType _value) {
-                return new CollectionType(ConjureCollectionType.SET, options.nonNullCollections());
+                if (options.nonNullCollections() && options.primitiveOptimizedCollections()) {
```

Comment:
Does `options.primitiveOptimizedCollections()` have any impact for sets?

Reply:
Yes, it will use newNonNullSet() instead of newSet(). Maybe not the most correct name in this case, but was going for wrapping all the codegen changes in this one flag.

Reply:
Functionally those two methods just call `new Set<>()`, but can be modified in the future with some validating set if necessary